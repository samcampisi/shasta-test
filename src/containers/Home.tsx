import { setUsername } from '../actions/usersActions';
import { BottomTabNavigationProp } from '@react-navigation/bottom-tabs';
import { RouteProp } from '@react-navigation/native';
import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Text, View, TextInput } from 'react-native';
import { AppTabParamList } from '../App';
import styles from '../styles/Home.style';
import Button from '../components/Button';
import { animate, enableAnimation } from '../lib/animation';

export interface HomeProps {
  username: string;
  navigation: BottomTabNavigationProp<AppTabParamList, 'Home'>;
  route: RouteProp<AppTabParamList, 'Home'>;
}

const Home = () => {
  const [userName, setUserName] = useState('');
  const [inputFocused, setInputFocused] = useState(false);
  const username = useSelector((state: any) => state.users.username);
  const dispatch = useDispatch();

  useEffect(() => {
    enableAnimation();
  }, []);

  useEffect(() => {
    animate();
  }, [username]);

  const onChangeText = (text: string) => {
    setUserName(text);
  };

  const onInputFocus = () => {
    setInputFocused(true);
  };

  const onInputBlur = () => {
    setInputFocused(false);
  };

  const onButtonPress = () => {
    dispatch(setUsername(userName));
    setUserName('');
    animate();
  };

  return (
    <View style={styles.container}>
      <Text style={styles.welcome}>Welcome{username ? `, ${username}` : ''}!</Text>
      {!username && (
        <View>
          <Text style={styles.instructions}>Please enter your username below:</Text>
          <TextInput
            style={[styles.input, inputFocused && styles.inputFocused]}
            onChangeText={onChangeText}
            value={userName}
            placeholder="paquita-salas"
            onFocus={onInputFocus}
            onBlur={onInputBlur}
            autoCapitalize="none"
          />
          <Button onPress={onButtonPress} title="Continue" disabled={!userName.length} />
        </View>
      )}
    </View>
  );
};

export default Home;
