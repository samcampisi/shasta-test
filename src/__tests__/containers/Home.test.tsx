import Home from '../../containers/Home';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
//import { render } from 'react-native-testing-library';
import { render, cleanup, store } from '../test-utils';

Enzyme.configure({ adapter: new Adapter() });

const createTestProps = (props?: object) => ({
  ...props,
});

jest.mock('../../../node_modules/react-native/Libraries/LayoutAnimation/LayoutAnimation.js');
jest.mock('../../actions/usersActions', () => {
  return {
    fetchUser: jest.fn().mockReturnValue('mock user action'),
  };
});

afterEach(() => {
  cleanup();
  store.clearActions();
});

describe('Home', () => {
  const navigate = jest.fn();
  const props: any = createTestProps({
    navigation: {
      navigate,
    },
  });

  describe('Basic Home', () => {
    const { getByText, toJSON } = render(<Home {...props} />);
    it('should render a welcome', () => {
      expect(getByText(/welcome/i)).toBeDefined();
    });

    it('should match snapshot', () => {
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
