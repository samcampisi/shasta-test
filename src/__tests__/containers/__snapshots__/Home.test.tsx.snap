// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Basic Home should match snapshot 1`] = `null`;
